cdap-watchdog/src/test/java/co/cask/cdap/logging/save/LogSaverPluginTest.java

```diff
@@ -240,6 +237,55 @@ public void testPlugin() throws Exception {
     }
   }
 
+  private void verifyLogSaverReset(LocationFactory locationFactory, List<LogEvent> allEvents,
+                                   LoggingContext loggingContext, String logBaseDir,
+                                   int i, long targetOffset, long mismatchOffset,
+                                   long expectedMetricsCount) throws Exception {
+    // Now reset the logSaver to start reading from reset offset
+    // Change the log meta table so that checkpoints are new, and old log files are not read during read later
+    LogSaverTableUtilOverride.setLogMetaTableName("log.meta" + i);
+
+    // Reset checkpoint for log saver plugin to Checkpoint{nextOffset=mismatchOffset, maxEventTime=<targetTime>}
+    // where <targetTime> is the timestamp of the event with targetOffset in Kafka,
+    // or Long.MAX_VALUE if targetOffset exceeds the largest index of allEvents, or 0 if targetOffset is -1.
+    // The KafkaOffsetFinder called in LogSaver should be able to discover the targetTime
+    // different from the timestamp of the message fetched with the mismatchOffset,
+    // and KafkaOffsetFinder should find the correct offset of the message with the same timestamp as the targetTime.
+    resetLogSaverPluginCheckpoint(allEvents, targetOffset, mismatchOffset);
+
+    // Change base dir so that new files get written and read from a different location (makes it easy for asserting)
+    String newLogBaseDir = logBaseDir + i;
+    cConf.set(LoggingConfiguration.LOG_BASE_DIR, newLogBaseDir);
+    locationFactory.create(namespaceDir).append("NS_1").append(newLogBaseDir);
+
+    // Start log saver again
+    startLogSaver();
+
+    FileMetaDataManager fileMetaDataManager = injector.getInstance(FileMetaDataManager.class);
+    if (targetOffset == Long.MAX_VALUE) {
+      // Sleep for 6 seconds to wait for LogSaver to start processing messages
+      Thread.sleep(6000);
```

Comment:
The last log message needs to be provided to `waitTillLogSaverDone` to verify whether LogSaver has finished. However, in this case, LogSaver doesn't have any log message to save, so `waitTillLogSaverDone` cannot be used